Problem_Set_2/hangman.py

```diff
@@ -0,0 +1,400 @@
+        print("Good guess:", display)
+      elif guess.isalpha() is False:
+        if warnings_left > 0:
+          warnings_left -= 1
+          print("Oops! That is not a valid letter. You have", warnings_left, "warnings left:", display)
+        else:
+          guesses_left -= 1
+          print("Oops! That is not a valid letter. You have no warnings left, so you lose one guess:", display)
+      elif guess.isalpha() is True and guess in letters_guessed:
+        if warnings_left > 0:
+          warnings_left -= 1
+          print("Oops! You've already guessed that letter. You have", warnings_left, "warnings left:", display)
+        else:
+          guesses_left -= 1
+          print("Oops! You've already guessed that letter. You have no warnings left, so you lose one guess:", display)
+      elif guess.isalpha() is True and guess not in letters_guessed and guess in consonants:
+        letters_guessed.append(guess)
+        display = get_guessed_word(secret_word, letters_guessed)
+        print("Oops! That letter is not in my word:", display)
+        guesses_left -= 1
+      elif guess.isalpha() is True and guess not in letters_guessed and guess in vowels:
+        letters_guessed.append(guess)
+        display = get_guessed_word(secret_word, letters_guessed)
+        print("Oops! That letter is not in my word:", display)
+        guesses_left -= 2
+      print("-------------")
+    # pass
+    if is_word_guessed(secret_word, letters_guessed) == True:
+      print("Congratulations, you won!")
+      # print(set(secret_word))
+      total_score = guesses_left * len(set(secret_word))
+      # print(total_score)
+      print("Your total score for this game is:", total_score)
+    elif guesses_left == 0 and is_word_guessed(secret_word, letters_guessed) == False:
+      print("Sorry, you ran out of guesses. The word was", secret_word)
+
+# hangman("hello")
+
+# When you've completed your hangman function, scroll down to the bottom
+# of the file and uncomment the first two lines to test
+#(hint: you might want to pick your own
+# secret_word while you're doing your own testing)
+
+
+# -----------------------------------
+
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the 
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise: 
+    '''
+    # FILL IN YOUR CODE HERE AND DELETE "pass"
+    # print(my_word)
+    my_word = my_word.replace(" ", "")
+    # print(my_word)
+    if len(my_word) != len(other_word):
+      return False
+    for i in range(len(my_word)):
+      # print(my_word[i], other_word[i])
+      if my_word[i].isalpha() == True and other_word[i].isalpha() == True and my_word[i] != other_word[i]:
```

Comment:
Why not do `if my_word[i] != other_word[i] or not (my_word[i].isalpha())`

Reply:
It doesn't work. This code is trying to say iterate over the characters of both my_word and other_word. If there is an alphabet character in my_word, check if it is the same as the character in other_word. If however, my_word does not have an alphabetic character and instead an underscore for example, ignore it and move on. If the alphabetic parts of my_word and other_word match, then return true else return false.